### PR TITLE
pallet-evm: add support for transaction-level create2

### PR DESCRIPTION
**Pull request by sorpaas**
This allows CREATE2 to be directly called as transactions. Previously this must be done through factory contracts.

### CODE REVIEW

**Comment by shawntabrizi** [comment]
Is it possible to start writing tests for the EVM pallet?

File: frame/evm/src/lib.rs [hunk 1]
```diff
@@ -378,6 +390,76 @@ decl_module! {
 
 			ret.map_err(Into::into)
 		}
+
+		/// Issue an EVM create2 operation.
+		#[weight = WeightForCallCreate]
```
Comment (kianenigma) [inline frame/evm/src/lib.rs]: You can use `weight::FunctionOf` here to make your life easier :D

Comment (sorpaas) [reply frame/evm/src/lib.rs]: Fixed!

**Comment by sorpaas** [comment]
@shawntabrizi Let me do it in a separate PR.

**Pull request merged**